src/Umbraco.Core/IO/MediaFileSystem.cs

```diff
@@ -15,58 +16,60 @@ public class MediaFileSystem : FileSystemWrapper
 	{
 	    private readonly IContentSection _contentConfig;
 
-	    public MediaFileSystem(IFileSystem wrapped)
+	    public MediaFileSystem(IFileSystem2 wrapped)
 			: this(wrapped, UmbracoConfig.For.UmbracoSettings().Content)
 		{
 		}
 
-        public MediaFileSystem(IFileSystem wrapped, IContentSection contentConfig) : base(wrapped)
+        public MediaFileSystem(IFileSystem2 wrapped, IContentSection contentConfig) : base(wrapped)
         {
             _contentConfig = contentConfig;
         }
 
-	    public string GetRelativePath(int propertyId, string fileName)
+        // none of the methods below are used in Core anymore
+
+        [Obsolete("This low-level method should NOT exist.")]
+        public string GetRelativePath(int propertyId, string fileName)
 		{
-            var seperator = _contentConfig.UploadAllowDirectories
+            var sep = _contentConfig.UploadAllowDirectories
 				? Path.DirectorySeparatorChar
 				: '-';
 
-			return propertyId.ToString(CultureInfo.InvariantCulture) + seperator + fileName;
+			return propertyId.ToString(CultureInfo.InvariantCulture) + sep + fileName;
 		}
 
+        [Obsolete("This low-level method should NOT exist.", false)]
         public string GetRelativePath(string subfolder, string fileName)
         {
-            var seperator = _contentConfig.UploadAllowDirectories
+            var sep = _contentConfig.UploadAllowDirectories
                 ? Path.DirectorySeparatorChar
                 : '-';
 
-            return subfolder + seperator + fileName;
+            return subfolder + sep + fileName;
         }
 
+        [Obsolete("Use ImageHelper.GetThumbnails instead.", false)]
 		public IEnumerable<string> GetThumbnails(string path)
-		{
-			var parentDirectory = Path.GetDirectoryName(path);
-			var extension = Path.GetExtension(path);
-
-			return GetFiles(parentDirectory)
-				.Where(x => x.StartsWith(path.TrimEnd(extension) + "_thumb") || x.StartsWith(path.TrimEnd(extension) + "_big-thumb"))
-				.ToList();
+        {
+            return ImageHelper.GetThumbnails(this, path);
 		}
 
-		public void DeleteFile(string path, bool deleteThumbnails)
+        [Obsolete("Use ImageHelper.DeleteFile instead.", false)]

```

Comment:
So I got things wrong. Most of the `ImageHelper` and `MediaHelper` work was meant to align `MediaFileSystem` with pure `IFileSystem` by removing specific stuff from `MediaFileSystem`. Now ok if we use that filesystem as a layer on top of `IFileSystem` to implement anything that's media-related... fine, that addresses some of your other comments, will move everything back into shape.